### DISCUSSION TITLE
Restructure project file structure

### DISCUSSION THREAD

**Discussion by MichelMichels** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3511] [Ideas]
## Change

I'd like to propose a change in file/directory structure of the project.

Current problems:
* On page load of the [GitHub project](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit), I have to scroll on a 15" laptop multiple time to even see the README.md
* A lot of files in the root folder

Proposition:
* I'd like to propose following structure by David Fowler: [GitHub gist link](https://gist.github.com/davidfowl/ed7564297c61fe9ab814)

This proposition would move all code related files to a `src` folder.

## Screenshots

![image](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/6505319/e30d7cbe-ea65-4678-8af5-bf175b0c2c97)

**Comment by MichelMichels** [comment]
I put together a branch that showcases the changes: [branch](https://github.com/MichelMichels/MaterialDesignInXamlToolkit/tree/feature/project-structure)

For further shrinkage, I'd also remove the `.slnf` files and the UWP solution.

**Reply by Keboo** [reply]
Awesome I will take a look. At first glance it does look like a good change; though I might want to get the PRs for the smart hint in first to avoid lots of conflicts.

**Reply by MichelMichels** [reply]
That's no problem. I'll try to keep this branch synced.
The only thing I'm not really quite sure about, are the github workflow files. This should definitely be reviewed.

**Comment by MichelMichels** [comment]
@Keboo would there be a valid reason why the project for resource generation is named `mdresgen`? This seems atypical for a C# project name. Also, I stumbled across `MaterialDesignResourceGeneration.slnf` in the root directory, is this an artifact of the previous name?

**Reply by Keboo** [reply]
The name `mdresgen` is a name that existed before I took over the project. I have considered renaming it many times and never got around to doing so. I am perfectly happy to see it get a better name.
